src/tools/run-make-support/src/lib.rs

```diff
@@ -295,6 +295,24 @@ pub fn not_contains<P: AsRef<Path>>(path: P, expected: &str) -> bool {
     !path.as_ref().file_name().is_some_and(|name| name.to_str().unwrap().contains(expected))
 }
 
+/// Builds a static lib (`.lib` on Windows MSVC and `.a` for the rest) with the given name.
+#[track_caller]
+pub fn build_native_static_lib(lib_name: &str) -> PathBuf {
+    let mut obj_file = format!("{lib_name}.o");
+    let src = format!("{lib_name}.c");
+    let lib_path = static_lib_name(lib_name);
+    if is_msvc() {
+        cc().arg("-c").out_exe(&obj_file).input(src).run();
+    } else {
+        cc().arg("-v").arg("-c").out_exe(&obj_file).input(src).run();
+    };
+    if is_msvc() {
+        obj_file = format!("{obj_file}.obj");
+    }
```

Comment:
I suppose this is a little questionable, as it creates a file named "something.o.obj". It is almost instantly dismissed by the following `ar`, though.

Reply:
Suggestion: maybe use [`PathBuf::set_extension`](https://doc.rust-lang.org/std/path/struct.PathBuf.html#method.set_extension) which would correctly handle changing `.o` -> `.obj`?

Reply:
Just gave it a try!